src/ui/components/Header/Header.tsx

```diff
@@ -178,25 +180,31 @@ export default function Header() {
 
   assert(recording != null);
 
-  let dashboardUrl = window.location.origin;
+  let fallbackUrl = window.location.origin;
   if (recording.workspace !== null) {
-    dashboardUrl = `/team/${recording.workspace?.id}`;
-
-    if (isTestSuiteReplay(recording) && recording.metadata?.test) {
-      const runId = getTestRunId(recording.metadata.test);
-      if (runId != null) {
-        dashboardUrl += `/runs/${runId}`;
-      }
+    fallbackUrl = `/team/${recording.workspace?.id}`;
+    if (isTestSuiteReplay(recording) && recording.testRun?.id) {
+      fallbackUrl += `/runs/${recording.testRun.id}`;
     }
   } else {
-    dashboardUrl = "/team/me/recordings";
+    fallbackUrl = "/team/me/recordings";
   }
 
+  const goBack = (e: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    e.preventDefault();
+
+    if (window.history.length > 1) {
+      router.back();
+    } else {
+      router.push(fallbackUrl);
+    }
+  };
```

Comment:
I was hoping we could do this but I don't think we can if we can't inspect the history (which, afaict next doesn't provide since it relies on browser state). If we added this and the user direct linked to a replay, clicking the in-app back button would take them out of replay instead of returning to the library.

Reply:
to be able to navigate back (to tests or test run) we can either
1. this approach that takes them out of replay
2. have a referer query param, that takes full path (encoded URI). for example referer=/team/team-id/whatever, and we use this as is.
3. have just tiny query param when linking a recording, for example from=tests, find and return testId from backend for that recording, and form a url for it similarly we do test run.

I think 3 takes more time to implement and is just very rigid and we can have other places we want to link a recording from. 2 should be fine IMO. What do you think?

Reply:
I think using the referrer would be okay. Another alternative would be to store the previous test suites path in context and use it here. I don't really like cramming more data elements into context though so the querystring param is probably better.